src/main/java/de/tum/cit/aet/artemis/quiz/service/QuizExerciseService.java

```diff
@@ -291,31 +293,41 @@ public void handleDndQuizFileUpdates(QuizExercise updatedExercise, QuizExercise
             }
         }
 
-        fileService.deleteFiles(filesToRemove.stream().map(Path::of).toList());
+        var allFilesToRemoveMerged = filesToRemove.entrySet().stream()
+                .flatMap(entry -> entry.getValue().stream().map(p -> FilePathService.fileSystemPathForExternalUri(URI.create(p), entry.getKey()))).filter(Objects::nonNull)
+                .toList();
+
+        fileService.deleteFiles(allFilesToRemoveMerged);
     }
 
-    private Set<String> getAllPathsFromDragAndDropQuestionsOfExercise(QuizExercise quizExercise) {
-        Set<String> paths = new HashSet<>();
+    private Map<FilePathType, Set<String>> getAllPathsFromDragAndDropQuestionsOfExercise(QuizExercise quizExercise) {
+        Map<FilePathType, Set<String>> paths = new HashMap<>();
+        paths.put(FilePathType.DRAG_AND_DROP_BACKGROUND, new HashSet<>());
+        paths.put(FilePathType.DRAG_ITEM, new HashSet<>());
+
         for (var question : quizExercise.getQuizQuestions()) {
             if (question instanceof DragAndDropQuestion dragAndDropQuestion) {
                 if (dragAndDropQuestion.getBackgroundFilePath() != null) {
-                    paths.add(dragAndDropQuestion.getBackgroundFilePath());
+                    paths.get(FilePathType.DRAG_AND_DROP_BACKGROUND).add(dragAndDropQuestion.getBackgroundFilePath());
                 }
-                paths.addAll(dragAndDropQuestion.getDragItems().stream().map(DragItem::getPictureFilePath).filter(Objects::nonNull).collect(Collectors.toSet()));
+                Set<String> dragItemPaths = dragAndDropQuestion.getDragItems().stream().map(DragItem::getPictureFilePath).filter(Objects::nonNull).collect(Collectors.toSet());
+                paths.get(FilePathType.DRAG_ITEM).addAll(dragItemPaths);
             }
         }
+
         return paths;
     }
 
-    private void handleDndQuestionUpdate(DragAndDropQuestion dragAndDropQuestion, Set<String> oldPaths, Set<String> filesToRemove, Map<String, MultipartFile> fileMap,
-            DragAndDropQuestion questionUpdate) throws IOException {
+    private void handleDndQuestionUpdate(DragAndDropQuestion dragAndDropQuestion, Map<FilePathType, Set<String>> oldPaths, Map<FilePathType, Set<String>> filesToRemove,
+            Map<String, MultipartFile> fileMap, DragAndDropQuestion questionUpdate) throws IOException {
         String newBackgroundPath = dragAndDropQuestion.getBackgroundFilePath();
 
         // Don't do anything if the path is null because it's getting removed
         if (newBackgroundPath != null) {
-            if (oldPaths.contains(newBackgroundPath)) {
+            var oldBackgroundPaths = oldPaths.get(FilePathType.DRAG_AND_DROP_BACKGROUND);
```

Comment:
I do not think this is a reasonable usage of var here
```suggestion
            Set<String> oldBackgroundPaths = oldPaths.get(FilePathType.DRAG_AND_DROP_BACKGROUND);
```